### PR TITLE
FEAT-#6890: Modin implementation of DataFrame API standard

### LINKED ISSUE #6890: Add consortium DataFrame standard entrypoint
For reference: https://github.com/pandas-dev/pandas/pull/54383

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6890 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/getting_started/installation.rst [hunk 1]
```diff
@@ -74,6 +74,15 @@ storage formats or for different functionalities of Modin. Here is a list of dep
 
   pip install "modin[mpi]" # If you want to use MPI through unidist execution engine
 
+
+Consortium Standard-compatible implementation based on modin
```
Comment (YarShev) [inline docs/getting_started/installation.rst:78]:
```suggestion
Consortium Standard-compatible implementation based on Modin
```

File: modin/tests/test_dataframe_api_standard.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+
+import modin.pandas
+
+
+def test_dataframe_consortium() -> None:
```
Comment (YarShev) [inline modin/tests/test_dataframe_api_standard.py:19]:
```suggestion
def test_dataframe_api_standard() -> None:
```
?

File: environment-dev.yml [hunk 3]
```diff
@@ -61,6 +61,7 @@ dependencies:
   - isort>=5.12
 
   - pip:
+      - git+https://github.com/data-apis/dataframe-api-compat.git@main
```
Comment (YarShev) [inline environment-dev.yml:64]: When do we expect a new release of dataframe-api-compat containing support for Modin?

Comment (anmyachev) [reply environment-dev.yml:64]: I expect that I will have time to make a new release of this package before the new release of Modin. As a last resort, we will limit it to some working commit.

**Review by YarShev — APPROVED** [review]

**Pull request merged**